contracts/exchangeIssuance/FlashMintLeveraged.sol

```diff
@@ -0,0 +1,1249 @@
+     *
+     * @param _collateralToken       Address of the collateralToken to withdraw from Aave lending pool
+     * @param _collateralAmount      Amount of collateralToken to withdraw
+     */
+    function _withdrawCollateralToken(
+        address _collateralToken,
+        uint256 _collateralAmount
+    ) internal {
+        LENDING_POOL.withdraw(_collateralToken, _collateralAmount, address(this));
+    }
+
+    /**
+     * Sets a max approval limit for an ERC20 token, provided the current allowance
+     * is less than the required allownce.
+     *
+     * @param _token              Token to approve
+     * @param _spender            Spender address to approve
+     * @param _requiredAllowance  Target allowance to set
+     */
+    function _safeApprove(
+        IERC20 _token,
+        address _spender,
+        uint256 _requiredAllowance
+    )
+        internal
+    {
+        uint256 allowance = _token.allowance(address(this), _spender);
+        if (allowance < _requiredAllowance) {
+            _token.safeIncreaseAllowance(_spender, MAX_UINT256 - allowance);
+        }
+    }
+
+    /**
+     * Approves max amount of token to lending pool
+     *
+     * @param _token              Address of the token to approve
+     */
+    function _approveTokenToLendingPool(
+        IERC20 _token
+    )
+    internal
+    {
+        uint256 allowance = _token.allowance(address(this), address(LENDING_POOL));
+        if (allowance > 0) {
+            _token.approve(address(LENDING_POOL), 0);
+        }
+        _token.approve(address(LENDING_POOL), MAX_UINT256);
+    }
+
+    /**
+     * Triggers the flashloan from the BalancerV2 Vault
+     *
+     * @param assets         Addresses of tokens to loan 
+     * @param amounts        Amounts to loan
+     * @param params         Encoded memory to forward to the executeOperation method
+     */
+    function _flashloan(
+        address[] memory assets,
+        uint256[] memory amounts,
+        bytes memory params
+    )
+    internal
+    {
+        require(flashLoanBenefactor == address(0), "Flashloan already taken");
+        flashLoanBenefactor = msg.sender;
+        balancerV2Vault.flashLoan(this, assets, amounts, params);
+        flashLoanBenefactor = address(0);
+    }
+
+    /**
+     * Redeems a given amount of SetToken.
+     *
+     * @param _setToken     Address of the SetToken to be redeemed
+     * @param _amount       Amount of SetToken to be redeemed
+     */
+    function _redeemExactSet(ISetToken _setToken, uint256 _amount) internal returns (uint256) {
+        _setToken.safeTransferFrom(msg.sender, address(this), _amount);
+        debtIssuanceModule.redeem(_setToken, _amount, address(this));
+    }
+
+    /**  Returns the absolute value of x
+    * @param x The number to take the absolute value of
+    */
+    function abs(int256 x) internal pure returns (uint256) {
+        return x >= 0 ? uint256(x) : uint256(-x);
+    }
+      receive() external payable {}
```

Comment:
nit: some weird formatting issue

Reply:
resolved in https://github.com/IndexCoop/index-coop-smart-contracts/pull/143/commits/bbc322357d4c4e919f6378fb22e36dbb97cc75a5